### PR TITLE
Replace freeport with independent impl.

### PR DESCRIPTION
**Pull request by andrewvc**
This removes a trivial dep I recently added and substitutes it for a small function.

### CODE REVIEW

File: libbeat/testing/available_port.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+package testing
+
+import "net"
+
+// AvailableTCPPort returns an unused TCP port for 127.0.0.1.
+func AvailableTCPPort() (uint16, error) {
+	resolved, err := net.ResolveTCPAddr("ip4", "127.0.0.1:0")
```
Comment (ruflin) [inline libbeat/testing/available_port.go]: Seems like this causes some troubles on CI.

File: libbeat/testing/available_port.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package testing
+
+import "net"
+
+// AvailableTCPPort returns an unused TCP port for 127.0.0.1.
+func AvailableTCPPort() (uint16, error) {
```
Comment (andrewkroh) [inline libbeat/testing/available_port.go]: Perhaps the name should include `TCP4`? You might want to add a TCP6 version in the future.

**Comment by andrewvc** [comment]
jenkins, please retest this

**Comment by andrewvc** [comment]
jenkins, retest this please

**Comment by ruflin** [comment]
Thanks for the cleanup.

**Pull request merged**